File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int p=0;

        int m=prices[0];
        int  j=prices.size();

        for(auto s:prices){
            p=max(p,s-m);
                m=min(m,s);
        }
        return p;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        i=0
        j=len(heights)-1
        area=-1
        while i<j:
            if heights[i]<=heights[j]:
                x=heights[i]*(j-i)
                area=max(area,x)
                i+=1
            else: 
                x=heights[j]*(j-i)
                area=max(area,x)
                j-=1
        return area
